### PR TITLE
case insensitive regex

### PR DESCRIPTION
**Pull request by treehousetim**
Using chart.js I noticed an exception was being thrown when I formatted my color string using all caps.  not sure why I used all caps but I did, and the chart was drawn correctly.  The tooltip was where the exception was being thrown so it was not immediately obvious what the problem was.  Did a little digging and discovered that if I added /i for case insensitive regex matching everything was happy again.

All I've tested is RGB and RGBA, but I added the /i to all 4 of the regexs.  the first two may be unnecessary.

### CODE REVIEW

**Comment by etimberg** [comment]
looks good to me. @simonbrunel will this impact anything you are doing?

**Comment by etimberg** [comment]
@tannerlinsley @derekperkins can you check this as well?

**Comment by simonbrunel** [comment]
Not at all, I'm working on the "color-new" repository which also uses the chartjs color-string.

File: color-string.js [hunk 1]
```diff
@@ -24,10 +24,10 @@ function getRgba(string) {
    if (!string) {
       return;
    }
-   var abbr =  /^#([a-fA-F0-9]{3})$/,
-       hex =  /^#([a-fA-F0-9]{6})$/,
-       rgba = /^rgba?\(\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/,
-       per = /^rgba?\(\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/,
+   var abbr =  /^#([a-fA-F0-9]{3})$/i,
```
Comment (simonbrunel) [inline color-string.js:27]: I think the /i is useless for `abbr` and `hex` because regex already include lower/upper cases (`a-fA-F`)

**Comment by simonbrunel** [comment]
I quickly read [W3C recommendations ](https://www.w3.org/TR/css3-color/#rgb-color), color names are case-insensitive, but color model notation looks case-sensitive:

> The format of an RGB value in the functional notation is ‘rgb(’ followed by a comma-separated

@treehousetim I would rather keep the current case-sensitive implementation and change `RGB(` for `rgb(` in your code since it doesn't look correct.

**Comment by treehousetim** [comment]
I respect the idea of aligning with standards.  However it seems the standard is ambiguous here.

The only explicit wording I can find on this subject is in the CSS2 spec:

```
All CSS syntax is case-insensitive within the ASCII range
(i.e., [a-z] and [A-Z] are equivalent), except for
parts that are not under the control of CSS. 
```

[This answer](http://stackoverflow.com/a/17968374/608781) on StackOverflow confirms this.

In the interest of flexibility, and out of consideration of the unhelpful error message from the downstream project (chart.js) I respectfully request you to reconsider.

**Comment by simonbrunel** [comment]
Agree, the standard is ambiguous and browsers accept case insensitive function notation.

**Comment by etimberg** [comment]
Hmm, I feel like we could easily improve this with the case insensitive regex. If the browsers support it too, I'd vote to do the same even if it's non-standard.

**Comment by treehousetim** [comment]
It's been a few months - any chance of this getting merged? :)

**Pull request merged**